### PR TITLE
Migrate PostBlock to type definitions

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Migrate PostBlock endpoint to use JsonTypeDefinition.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java [hunk 1]
```diff
@@ -400,7 +400,7 @@ private void addBeaconHandlers(final DataProvider dataProvider) {
     app.get(GetBlockHeaders.ROUTE, new GetBlockHeaders(dataProvider, jsonProvider));
     app.get(GetBlockHeader.ROUTE, new GetBlockHeader(dataProvider, jsonProvider));
 
-    app.post(PostBlock.ROUTE, new PostBlock(dataProvider, jsonProvider));
+    addMigratedEndpoint(new PostBlock(schemaCache, dataProvider));
```
Comment (tbenr) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java]: nit: all the others constructor seems to have dataProvider as first param. Might be good to have them all aligned.

File: data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/migrated/paths/_eth_v1_beacon_blinded_blocks.json [hunk 2]
```diff
@@ -42,7 +42,13 @@
...
       "503" : {
         "description" : "Beacon node is currently syncing.",
-        "content" : { }
+        "content" : {
+          "application/json" : {
+            "schema" : {
+              "$ref" : "#/components/schemas/HttpErrorResponse"
+            }
+          }
+        }
```
Comment (tbenr) [inline data/beaconrestapi/src/integration-test/resources/tech/pegasys/teku/beaconrestapi/beacon/migrated/paths/_eth_v1_beacon_blinded_blocks.json:51]: we should add a test to cover post blinded block

**Review by tbenr — APPROVED** [review]
LGTM. Just a couple of things

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java]: agreed, i'll make the change.

**Pull request merged**